### PR TITLE
Added vimls

### PR DESCRIPTION
**Pull request by mjlbach**
Probably needs some cleanup, pertains to #66

### CODE REVIEW

File: lua/nvim_lsp/vimls.lua [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            vimruntime = "",
+            runtimepath = "",
+            diagnostic = {
+                enable = true
+            },
+            indexes = {
+                runtimepath = true,
+                gap = 100,
+                count = 3,
+                projectRootPatterns = {"nvim", ".git", "autoload", "plugin"}
```
Comment (justinmk) [inline lua/nvim_lsp/vimls.lua]: "runtime" would be useful too I suppose (first in the list)

**Comment by justinmk** [comment]
please use 2-space indent to match the rest of the project.

File: lua/nvim_lsp/vimls.lua (hunk 1 above)
Comment (mjlbach) [reply lua/nvim_lsp/vimls.lua]: Fixed

**Comment by mjlbach** [comment]
> please use 2-space indent to match the rest of the project.

Downside of adding a lua autoformatter to my local repo and not checking the config file, fixed the settings.

**Comment by justinmk** [comment]
@mjlbach ooh, what lua autoformatter are you using? I could use one... :)

**Comment by mjlbach** [comment]
> @mjlbach ooh, what lua autoformatter are you using? I could use one... :)

I've been shopping around and settled on [Koihik/LuaFormatter](https://github.com/Koihik/LuaFormatter)! It's used in the top rated lua formatting plugin for vscode, so I repurposed it 😇

**Comment by norcalli** [comment]
I also use LuaFormatter, but I haven't customized/tuned it perfectly yet to what our standards are. It works fine on the `formatexpr`, though.

I was considering trying to make a formatter from the treesitter api.

**Pull request merged**